### ISSUE TITLE
Google Login for first Auth: "The requested action is invalid."

### ISSUE THREAD

**Issue by broldfield** [revisit-studies/study#384]
**Describe the bug**
Background:
I encountered this bug on the 21st May 2024, I haven't merged my fork since then.
When following the tutorial, after completing the 'Authentication' step and attempting to add the first user as the Admin via Study browser > Settings, the login popup returns the error "The requested action is invalid.". When inspecting the popup with the browsers devtools, the console contains the error "Request is missing required data". I tried different browsers, computers and disabling adblockers.

**Potential Fix:**
Forewarning in that I know next to nothing about Firebase, so this may not be the best solution.
I fixed this issue by adding a resolver to the function `signInWithPopup()` in `./src/Login.tsx`. Specifically, I:

```
// Added an import of browserPopupRedirectResolver from @firebase/auth
import {
  getAuth, signInWithPopup, GoogleAuthProvider,
  browserPopupRedirectResolver,
} from '@firebase/auth';

//In the signInWithGoogle() function block
// Added browserPopupRedirectResolver to signInWithPopup() as the third param.
await signInWithPopup(auth, provider, browserPopupRedirectResolver);
```

An example can be seen here: https://github.com/broldfield/revisit/blob/main/src/Login.tsx

After adding this, the Auth begun working as expected. I haven't tried reproducing the bug after adding in the fix as I don't want to mess with `user-management` in firebase.

**Comment by JackWilb** [comment]
Hey @broldfield! Thanks for reaching out and reporting the issue and for suggesting a fix. We’ll take a look and see if we can figure out what the issue is. Just for our information, would you be able to provide your browser info (and any extensions you might be using) so we can try to reproduce the problem locally?

**Comment by broldfield** [comment]
When I was debugging I tried both Chrome and Microsoft Edge on two computers. 
Chrome: 
Ublock Origin/Privacy Badger. Tried with both enabled and disabled.
Version 125.0.6422.77 (Official Build) (64-bit)

Microsoft Edge:
Ublock Origin. Tried with enabled and disabled.
Version 125.0.2535.67 (Official build) (64-bit)

**Comment by JackWilb** [comment]
After reading through the documentation, it seems that the `browserPopupRedirectResolver ` helps with consistency across browsers. I'm not sure why it's not better documented by Google, but thanks for helping us find it. I've pushed a commit into https://github.com/revisit-studies/study/pull/387 that applies your suggested change. Thanks again for the report! I'll close this once we merge the PR to `dev`.

**Comment by JackWilb** [comment]
Closed via #387
